plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/ImplementAvgDecimal.java

```diff
@@ -0,0 +1,72 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcExpression;
+import io.trino.plugin.jdbc.expression.AggregateFunctionRule;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.Variable;
+import io.trino.spi.type.DecimalType;
+
+import java.util.Optional;
+
+import static com.google.common.base.Verify.verify;
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.expressionType;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.singleInput;
+import static io.trino.plugin.jdbc.expression.AggregateFunctionPatterns.variable;
+import static java.lang.String.format;
+
+/**
+ * Implements {@code avg(decimal(p, s)}
+ */
+public class ImplementAvgDecimal
+        implements AggregateFunctionRule
+{
+    private static final Capture<Variable> INPUT = newCapture();
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().equalTo("avg"))
+                .with(singleInput().matching(
+                        variable()
+                                .with(expressionType().matching(DecimalType.class::isInstance))
+                                .capturedAs(INPUT)));
+    }
+
+    @Override
+    public Optional<JdbcExpression> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context)
+    {
+        Variable input = captures.get(INPUT);
+        JdbcColumnHandle columnHandle = (JdbcColumnHandle) context.getAssignment(input.getName());
+
+        DecimalType type = (DecimalType) columnHandle.getColumnType();
+        verify(aggregateFunction.getOutputType().equals(type));
+
+        String column = context.getIdentifierQuote().apply(columnHandle.getColumnName());
+
+        // wait https://issues.apache.org/jira/browse/IGNITE-14948 to be solved.
+        return Optional.of(new JdbcExpression(
+                format("CAST(sum(%s) / count(%s) AS decimal(%s, %s))", column, column, type.getPrecision() + 1, type.getScale()),
```

Comment:
What's the max precision supported by Ignite? If we get a DECIMAL(38, 0) will Ignite fail because it tries to cast to DECIMAL(39, 0).

Also, the return type of this doesn't match the input types. Maybe you should wrap the expression in a `round` to the original decimal scale and precision.

Reply:
> What's the max precision supported by Ignite? If we get a DECIMAL(38, 0) will Ignite fail because it tries to cast to DECIMAL(39, 0).
>
Ignite max precision is 39, so it's safe just expand 1 precision.
 
> Also, the return type of this doesn't match the input types. Maybe you should wrap the expression in a `round` to the original decimal scale and precision.
Actually, Ignite just act like it has already round for user, we do not need added round again.See the test in BaseJdbcConnectorTest#testNumericAggregationPushdown. It will test the correctness about the avg on the decimal.